### PR TITLE
Add more idata attributes for JAX samplers

### LINKED ISSUE #7262: Missing idata attributes from jax/numba samplers
### Description

```python
import pymc as pm

with pm.Model() as m:
    x = pm.Normal("x")
    idata1 = pm.sample(nuts_sampler="pymc")
    idata2 = pm.sample(nuts_sampler="numpyro")

print(idata1.posterior.attrs)
# {'created_at': '2024-04-17T08:06:27.074686',
#  'arviz_version': '0.17.1',
#  'inference_library': 'pymc',
#  'inference_library_version': '5.10.4',
#  'sampling_time': 1.4884159564971924,
#  'tuning_steps': 1000}
print(idata2.posterior.attrs)
# {'created_at': '2024-04-17T08:06:43.592711', 'arviz_version': '0.17.1'}
```

Would be nice to save those

### PR DESCRIPTION
**Pull request by osyuksel**
## Description

This PR provides a fix for [#7262](https://github.com/pymc-devs/pymc/issues/7262) by filling in missing `post_ attributes` keyword argument.

## Related Issue

- [x] Closes #7262

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [x] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

File: pymc/sampling/jax.py [hunk 1]
```diff
@@ -669,9 +669,7 @@ def sample_jax_nuts(
     else:
         log_likelihood = None
 
-    attrs = {
-        "sampling_time": (tic2 - tic1).total_seconds(),
-    }
+    attrs = {"sampling_time": (tic2 - tic1).total_seconds(), "tuning_steps": tune}
```
Comment (ricardoV94) [inline pymc/sampling/jax.py]: Add trailing comma so pre-commit formats this into one entry per line
```suggestion
    attrs = {"sampling_time": (tic2 - tic1).total_seconds(), "tuning_steps": tune,}
```

File: pymc/sampling/mcmc.py [hunk 2]
```diff
@@ -334,9 +334,7 @@ def _sample_external_nuts(
             default_dims=[],
         )
         attrs = make_attrs(
-            {
-                "sampling_time": t_sample,
-            },
+            {"sampling_time": t_sample, "tuning_steps": tune},
```
Comment (ricardoV94) [inline pymc/sampling/mcmc.py]: Same here

**Comment by osyuksel** [comment]
Added trailing commas for formatting.

**Review by twiecki — APPROVED** [review]

File: tests/sampling/test_mcmc_external.py [hunk 3]
```diff
@@ -44,6 +44,10 @@ def test_external_nuts_sampler(recwarn, nuts_sampler):
         idata1 = sample(**kwargs)
         idata2 = sample(**kwargs)
```
Comment (ricardoV94) [inline tests/sampling/test_mcmc_external.py:45]: Not introduced by you, but why are we sampling twice, is idata2 used anywhere?

File: tests/sampling/test_mcmc_external.py [hunk 4]
```diff
@@ -64,8 +68,11 @@ def test_external_nuts_sampler(recwarn, nuts_sampler):
     assert "L" in idata1.observed_data
     assert idata1.posterior.chain.size == 2
     assert idata1.posterior.draw.size == 500
+    assert idata1.posterior.tuning_steps == 500
     np.testing.assert_array_equal(idata1.posterior.x, idata2.posterior.x)
```
Comment (osyuksel) [inline tests/sampling/test_mcmc_external.py:72]: @ricardoV94 `idata2` is used in this line only, to check if the sampling is deterministic I think?

Comment (ricardoV94) [reply tests/sampling/test_mcmc_external.py:72]: Ah okay! Thanks for checking

**Pull request merged**